usl_models/usl_models/atmo_ml/model.py

```diff
@@ -1,22 +1,143 @@
 """AtmoML model definition."""
 
-from typing import TypeAlias, TypedDict
-
+import logging
+from typing import TypeAlias, TypedDict, List, Callable
+import keras
 import tensorflow as tf
 from keras import layers
-
-# from usl_models.atmo_ml import constants
+from usl_models.atmo_ml import constants
 from usl_models.atmo_ml import data_utils
 from usl_models.atmo_ml import model_params
 
 AtmoModelParams: TypeAlias = model_params.AtmoModelParams
 
 
-class AtmoInput(TypedDict):
-    """Input tensors dictionary for the Atmo model."""
+class AtmoModel:
+    """Atmo model class."""
+
+    class Input(TypedDict):
+        """Input tensors dictionary."""
+
+        spatial: tf.Tensor
+        spatiotemporal: tf.Tensor
+
+    class Result(TypedDict):
+        """Prediction result dictionary."""
+
+        prediction: tf.Tensor
+        chunk_id: str | tf.Tensor
+
+    def __init__(
+        self,
+        params: AtmoModelParams | None = None,
+        spatial_dims: tuple[int, int] = (constants.MAP_HEIGHT, constants.MAP_WIDTH),
+        num_spatial_features: int = constants.num_spatial_features,
```

Comment:
@Katsutoshii I have added these and some other changes, if you would like to check before merging